cmd/prefetchfiles-nri-plugin/main.go

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (c) 2023. Nydus Developers. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/containerd/nydus-snapshotter/pkg/errdefs"
+	"io"
+	"net"
+	"net/http"
+
+	"strings"
+
+	"log/syslog"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+
+	"github.com/containerd/nri/pkg/api"
+	"github.com/containerd/nri/pkg/stub"
+	"github.com/containerd/nydus-snapshotter/pkg/fanotify"
+	"github.com/containerd/nydus-snapshotter/version"
+	"github.com/pelletier/go-toml"
+)
+
+const (
+	defaultEvents     = "RunPodSandbox"
+	defaultServerPath = "/usr/local/bin/optimizer-server"
+	defaultPersistDir = "/opt/nri/optimizer/results"
+)
+
+type PluginConfig struct {
+	Events     []string `toml:"events"`
+	ServerPath string   `toml:"server_path"`
+	PersistDir string   `toml:"persist_dir"`
+	Timeout    int      `toml:"timeout"`
+	Overwrite  bool     `toml:"overwrite"`
+}
+
+type PluginArgs struct {
+	PluginName   string
+	PluginIdx    string
+	PluginEvents string
```

Comment:
If you only need to use `RunPodSandbox` interface, you can remove this configuration entry and define it as a constant.

Reply:
In order to improve the flexibility of prefetch files transmit, it may be necessary to increase the monitoring of the Create Container event, so I have temporarily kept this part. If any modifications are needed, I will make them the next time I update the PR.

Reply:
Only some specific events can catch the container image name & prefetch list, why do we still need to export the arg?